openmetadata-ui/src/main/resources/ui/src/constants/regex.constants.test.ts

```diff
@@ -0,0 +1,91 @@
+  it('EntityName regex should pass for the valid entity name', () => {
+    // Contains letters, numbers, and no special characters.
+    expect(ENTITY_NAME_REGEX.test('HelloWorld123')).toEqual(true);
+
+    // Contains letters, an underscore, and no special characters.
+    expect(ENTITY_NAME_REGEX.test('Cypress_Test')).toEqual(true);
+
+    // Contains letters and a hyphen.
+    expect(ENTITY_NAME_REGEX.test('My-Website')).toEqual(true);
+
+    // Contains letters, a period, and an uppercase letter.
+    expect(ENTITY_NAME_REGEX.test('Open.AI')).toEqual(true);
+
+    // Contains letters, a space, and an apostrophe.
+    expect(ENTITY_NAME_REGEX.test("John's Cafe")).toEqual(true);
+
+    // Contains letters, a space, an ampersand, and a period.
+    expect(ENTITY_NAME_REGEX.test('ACME & Co.')).toEqual(true);
+
+    // Contains letters, spaces, opening and closing parentheses.
+    expect(ENTITY_NAME_REGEX.test('This (is) a test')).toEqual(true);
+
+    // Contains Spanish characters
+    expect(ENTITY_NAME_REGEX.test('Buenos días')).toEqual(true);
+    expect(ENTITY_NAME_REGEX.test('Cómo estás')).toEqual(true);
+    expect(ENTITY_NAME_REGEX.test('Háblame en español')).toEqual(true);
+    expect(ENTITY_NAME_REGEX.test('Gracias')).toEqual(true);
+    expect(ENTITY_NAME_REGEX.test('Hola mundo')).toEqual(true);
+    expect(ENTITY_NAME_REGEX.test('áéíóú ÁÉÍÓÚ')).toEqual(true);
+
+    // Contains Russian characters
+    expect(ENTITY_NAME_REGEX.test('Привет мир')).toEqual(true);
+
+    // Contains Greek characters
+    expect(ENTITY_NAME_REGEX.test('Γειά σου κόσμε')).toEqual(true);
+
+    // Contains Arabic characters
+    expect(ENTITY_NAME_REGEX.test('مرحبا العالم')).toEqual(true);
+
+    // Contains Hebrew characters
+    expect(ENTITY_NAME_REGEX.test('שלום עולם')).toEqual(true);
+
+    // Contains Chinese characters
+    expect(ENTITY_NAME_REGEX.test('你好世界')).toEqual(true);
+
+    // Contains Korean characters
+    expect(ENTITY_NAME_REGEX.test('안녕하세요 세상')).toEqual(true);
+
+    // Contains Japanese characters
+    expect(ENTITY_NAME_REGEX.test('こんにちは世界')).toEqual(true);
+  });
+
+  it('EntityName regex should fail for the invalid entity name', () => {
+    // Contains letters, numbers, and # special characters.
+    expect(ENTITY_NAME_REGEX.test('HelloWorld123#')).toEqual(false);
+
+    // Contains letters, numbers, and $ special characters.
+    expect(ENTITY_NAME_REGEX.test('HelloWorld123$')).toEqual(false);
+
+    // Contains letters, numbers, and ! special characters.
+    expect(ENTITY_NAME_REGEX.test('HelloWorld123!')).toEqual(false);
+
+    // Contains letters, numbers, and @ special characters.
+    expect(ENTITY_NAME_REGEX.test('HelloWorld123@')).toEqual(false);
+
+    // Contains letters, numbers, and * special characters.
+    expect(ENTITY_NAME_REGEX.test('HelloWorld123*')).toEqual(false);
+
+    // Contains letters, numbers, and special characters.
+    expect(ENTITY_NAME_REGEX.test('!@#$%^&*()')).toEqual(false);
```

Comment:
% will be allowed for name, Let's confirm this

Reply:
can we add "." as well

Reply:
@harshach , "." is already there, the above one is for the test case.

Reply:
@chirag-madlani , for now we are just allowing % in glossary and glossaryTerms name, ref : https://github.com/open-metadata/OpenMetadata/pull/12382

I think that we can handle it separately after [this](https://github.com/open-metadata/OpenMetadata/pull/12382) PR got merged.